src/content/language/collections.md

```diff
@@ -242,78 +250,180 @@ final constantMap = const {2: 'helium', 10: 'neon', 18: 'argon'};
+
+You can use an `if` statement inside of a collection to
+conditionally include values in the collection based on a
+boolean expression. An `if` statement has this signature in
+a collection:
 
-Here's an example of using **collection if**
-to create a list with three or four items in it:
+```dart
+// boolean expression must be true to include result
+if (<bool_expression>) <result>
+```
 
-<?code-excerpt "misc/test/language_tour/built_in_types_test.dart (list-if)"?>
 ```dart
-var nav = ['Home', 'Furniture', 'Plants', if (promoActive) 'Outlet'];
+// boolean expression must be false to include result
+if (!<bool_expression>) <result>
 ```
 
-Dart also supports [if-case][] inside collection literals:
+In the following example, `1` is included in `items`
+because `includeItem` evaluates as true, but `2` is not
+included because `!includeItem` evaluates as false.
 
 ```dart
-var nav = ['Home', 'Furniture', 'Plants', if (login case 'Manager') 'Inventory'];
+var includeItem = true;
+var items = [0, if (includeItem) 1, if (!includeItem) 2, 3]; // [0, 1, 3]
 ```
 
-Here's an example of using **collection for**
-to manipulate the items of a list before
-adding them to another list:
+To learn more about the `if` statement, see
+[`if` statement][].
+
+[`if` statement]: /language/branches#if
+
+### If case statements
+
+You can use an `if-case` statement inside of a collection to
+conditionally include values in the collection if the
+matching expressions evaluate as true. An `if-case`
+statement has this signature in a collection:
 
-<?code-excerpt "misc/test/language_tour/built_in_types_test.dart (list-for)"?>
 ```dart
-var listOfInts = [1, 2, 3];
-var listOfStrings = ['#0', for (var i in listOfInts) '#$i'];
-assert(listOfStrings[1] == '#1');
+if (<expression> case expression_literal) <result>
 ```
 
-For more details and examples of using collection `if` and `for`, see the
-[control flow collections proposal.][collections proposal]
+In the following example, `1` is included because
+`a == 'one'` evaluates as true but `2` is not
+included because `b == 'two'` evaluates as false.
 
-[collections]: /libraries/dart-core#collections
-[type inference]: /language/type-system#type-inference
-[`List`]: {{site.dart-api}}/dart-core/List-class.html
-[`Map`]: {{site.dart-api}}/dart-core/Map-class.html
-[Using constructors]: /language/classes#using-constructors
-[collections proposal]: {{site.repo.dart.lang}}/blob/main/accepted/2.3/control-flow-collections/feature-specification.md
-[spread proposal]: {{site.repo.dart.lang}}/blob/main/accepted/2.3/spread-collections/feature-specification.md
-[generics]: /language/generics
-[`Set`]: {{site.dart-api}}/dart-core/Set-class.html
-[if-case]: /language/branches#if-case
+<?code-excerpt "misc/lib/language_tour/misc/if_case_operator_in_collection_a.dart (code_sample)"?>
+```dart
+var a = 'one';
+var b = '';
+var items = [
+  0,
+  if (a case 'one') 1,
+  if (b case 'two') 2,
+  3,
+  4,
+]; // [0, 1, 3, 4]
+```
```

Comment:
It might be worth showing an example that you couldn't do just using `if (a == 'one')`. Perhaps:

```dart
var string = 'hello';
var coordinates = [
  1,
  if (string case String(length: var length)) length,
  3,
]; // 1, 5, 3
```

That's a pretty weird, contrived example, though. It's hard to come up with a good but simple example. :-/

Reply:
Added!